includes/misc-functions.php

```diff
@@ -1274,9 +1274,644 @@ function give_set_upgrade_complete( $upgrade_action = '' ) {
+		remove_action( 'give_update_payment_status', 'give_donation_import_insert_default_payment_note', 1 );
+		remove_filter( 'give_insert_payment_args', 'give_donation_import_give_insert_payment_args', 11 );
+		remove_filter( 'give_update_donor_information', 'give_donation_import_update_donor_information', 11 );
+		remove_action( 'give_insert_payment', 'give_import_donation_insert_payment', 11 );
+
+		if ( $payment ) {
+
+			$report['create_donation'] = ( ! empty( $report['create_donation'] ) ? ( absint( $report['create_donation'] ) + 1 ) : 1 );
 
+			update_post_meta( $payment, '_give_payment_import', true );
+
+			if ( ! empty( $import_setting['csv'] ) ) {
+				update_post_meta( $payment, '_give_payment_import_id', $import_setting['csv'] );
+			}
+
+			// Insert Notes.
+			if ( ! empty( $data['notes'] ) ) {
+				give_insert_payment_note( $payment, $data['notes'] );
+			}
+
+			$meta_exists = array_keys( $raw_key, 'post_meta' );
+			if ( ! empty( $main_key ) && ! empty( $meta_exists ) ) {
+				foreach ( $meta_exists as $meta_exist ) {
+					if ( ! empty( $main_key[ $meta_exist ] ) && ! empty( $row_data[ $meta_exist ] ) ) {
+						update_post_meta( $payment, $main_key[ $meta_exist ], $row_data[ $meta_exist ] );
+					}
+				}
+			}
+		} else {
+			$report['failed_donation'] = ( ! empty( $report['failed_donation'] ) ? ( absint( $report['failed_donation'] ) + 1 ) : 1 );
+		}
+	}
+
+	// update the report
+	give_import_donation_report_update( $report );
+}
+
+/**
+ * Alter donor information when importing donations from CSV
+ *
+ * @since 1.8.13
+ *
+ * @param $donor
+ * @param $payment_id
+ * @param $payment_data
+ *
+ * @return Give_Donor
+ */
+function give_donation_import_update_donor_information( $donor, $payment_id, $payment_data ) {
+	$old_donor = $donor;
+	if ( ! empty( $payment_data['donor_id'] ) ) {
+		$donor_id = absint( $payment_data['donor_id'] );
+		$donor    = new Give_Donor( $donor_id );
+		if ( ! empty( $donor->id ) ) {
+			return $donor;
+		}
+	}
+
+	return $old_donor;
+}
+
+/*
+ * Give update purchase_count of give customer.
+ *
+ * @since 1.8.13
+ */
+function give_import_donation_insert_payment( $payment_id, $payment_data ) {
+	// Update Give Customers purchase_count
+	if ( ! empty( $payment_data['status'] ) && ( 'complete' === (string) $payment_data['status'] || 'publish' === (string) $payment_data['status'] ) ) {
+		$donor_id = (int) get_post_meta( $payment_id, '_give_payment_customer_id', true );
+		if ( ! empty( $donor_id ) ) {
+			$donor = new Give_Donor( $donor_id );
+			$donor->increase_purchase_count();
+		}
+	}
+}
+
+/**
+ * Add author id in in donation post
+ *
+ * @since 8.1.13
+ */
+function give_donation_import_give_insert_payment_args( $args, $payment_data ) {
+	if ( ! empty( $payment_data['user_info']['id'] ) ) {
+		$args['post_author'] = (int) $payment_data['user_info']['id'];
+	}
+
+	return $args;
+}
+
+/**
+ * Check if Import donation is duplicate
+ *
+ * Ref: https://xwp.github.io/engineering-best-practices/php/
```

Comment:
@raftaar1191 Why did you add this

Reply:
The argument that is being passed to WP query will result in better performance of MYSQL

So I thought of putting a referral link from which source did I get this will be useful

Reply:
Remove this one.

Reply:
Done